### ISSUE TITLE
Spoiler in part of sentence

### ISSUE THREAD

**Issue by Xeunyx-Cypher** [LemmyNet/lemmy-ui#687]

**Comment by ZeusOfTheCrows** [comment]
just want to add that reddit's spoiler tags, as far as i can see, are not keyboard accessible (and i don't know how well they do with screen readers)

**Comment by beedell-roke** [comment]
I'd much prefer non-singleton `spoiler` tags, [^1] [^2] when compared to `>!!<`, since they'd be a lot more easily readable, because non-text tags are parsed slightly differently by the HTML parser. An example is `<!--` and `-->`: [`stackoverflow.com/questions/442786`](https://stackoverflow.com/questions/442786/are-nested-html-comments-possible#comment27159111_12102131) explains.

[^1]: [`meta.stackexchange.com/q/69573`](https://meta.stackexchange.com/q/69573/623996)

[^2]: [`help.disqus.com/en/articles/1717306`](https://help.disqus.com/en/articles/1717306-spoiler-tags)

**Comment by ChaoticNeutralCzech** [comment]
> I'd much prefer
> 
> ```
> <spoiler></spoiler>
> ```
> 
> tags like Disqus has compared to `>!!<`, since they'd be a lot more easily readable, because non-text tags are parsed slightly differently by the HTML parser

Lemmy uses Markdown for comments. Why bring HTML syntax into this? Also, why are you feeding unsanitized Lemmy comments into an HTML parser?<script>alert("XSS")</script>

**Comment by rokejulianlockhart** [comment]
@ChaoticNeutralCzech, doesn't Lemmy also support HTML? I've rarely had instances where a Markdown implementation wasn't implemented as a superset of HTML.

**Comment by TeoTwawki** [comment]
I found this `||spoiler text||` to be a much more palatable format. just my 2 cents.

**Comment by Celestine20** [comment]
Now anyone working on this feature would be very cool <3
Thank you all the devs associated with this amazing project

**Comment by DraconicNEO** [comment]
@biosfood When you said that the feature was inplemented in your PR did you mean this Spoiler feature or something different?

**Comment by biosfood** [comment]
@DraconicNEO  I am sory for the misunderstanding, The PR added support for markdown inside the title of a spoiler, not the shorthand so this issue should stay open.

**Comment by DraconicNEO** [comment]
> @DraconicNEO I am sory for the misunderstanding, The PR added support for markdown inside the title of a spoiler, not the shorthand so this issue should stay open.

Ah I see, thanks for clarifying.

**Comment by DraconicNEO** [comment]
> As someone who will sometimes write the markdown format while I type, I definitely prefer the `>!spoiler!<` method that is more consistent with the way formatting is achieved for bold, italics, etc. Switching to HTML for one formatting option is confusing to the user, IMO.

I agree, it makes it easier to selectively blank out parts of a sentence or post which are spoilers, the drop down method is better for whole paragraphs but not for marking sentences or parts of sentences as spoilers.

**Comment by beedell-roke** [comment]
> [`#issuecomment-2035550744`](https://github.com/LemmyNet/lemmy-ui/issues/687#issuecomment-2035550744)

@DraconicNEO, merely having utilised the HTML `<spoiler>` tag instead doesn't mean that the `<details>` drop-down syntax (`display: block;`) would be utilised, just as utilizing custom `>!spoiler!<` syntax would inherently mean that it utilises `display: inline;`. I would, indeed, prefer, too, for Lemmy to set `spoiler {display: inline;}`. I see little reason for custom features to not adhere to at least standard markup syntax, considering that it makes it more portable.

**Comment by DraconicNEO** [comment]
> > [#687 (comment)](https://github.com/LemmyNet/lemmy-ui/issues/687#issuecomment-2035550744)
> 
> @DraconicNEO, merely having utilizing the HTML `<spoiler>` tag instead doesn't mean that the `<details>` drop-down syntax be utilized (`display: block;`) just as utilizing custom `>!spoiler!<` syntax would inherently mean that it utilizes `display: inline;`. I would indeed prefer too for Lemmy to set:
> 
> ```css
> spoiler {display: inline;}
> ```
> 
> I see little reason for custom features to not adhere to at least standard markup syntax, considering that it makes it more portable.

When I said drop down I worded it wrong, I meant this:
![image](https://github.com/LemmyNet/lemmy-ui/assets/44508470/08cb181b-e9c4-4e6b-9fd2-142a213568e8)

The one already used in lemmy-ui

**Comment by beedell-roke** [comment]
> [`#issuecomment-2036011353`](https://github.com/LemmyNet/lemmy-ui/issues/687#issuecomment-2036011353)

Yeah, @DraconicNEO, I hate that too. It's like they just create a `<details>` tag with the `<summary>` set to “Spoiler text” by default.

**Comment by dessalines** [comment]
There is currently no standardization of spoiler tags in commonmark: https://talk.commonmark.org/search?q=spoiler , and reddit has their own unique spoiler markup that we don't follow.

We're currently following [markdown-it 's recommended way of doing spoilers](https://github.com/markdown-it/markdown-it-container), that's also extensible to other types of content.

**Comment by IndefiniteBen** [comment]
I personally don't care much (I do prefer something easy to type on a touch keyboard) about the exact syntax, but this issue is about the lack of any way to include an inline spoiler. 

Do any of those search results (like https://github.com/traPtitech/markdown-it-spoiler/) support inline spoilers? From the discussion so far I get the impression this issue should be closed as "won't fix", unless one of those search results adds support for inline spoilers.

**Comment by DraconicNEO** [comment]
>There is currently no standardization of spoiler tags in commonmark: https://talk.commonmark.org/search?q=spoiler , and reddit has their own unique spoiler markup that we don't follow.

Is there any reason we couldn't use it or maybe do something similar just with different symbols? I mean I've seen a few other sites use that type of spoiler syntax. 
Also I don't think if we did do it that we should get rid of the current one, and instead we could have both together, as the current one is still useful for some situations and applications.

**Comment by DraconicNEO** [comment]
> @IndefiniteBen the one you linked works with inline spoilers.
> 
> I'm also not against us using any style, as long as _at least a few_ platforms use it. I don't want to start yet another competing markup just because of commonmark's slowness on picking one. I'd be interested to know in addition to reddit's inline spoiler markup, what other platforms like Discourse, Discord, and any others anyone can think of use.

Misskey and many misskey instances also have a [blur option](https://misskey-hub.net/en/docs/for-users/features/mfm/#blur) as a form of inline spoiler, not sure how widespread its use is though. 

I'm personally for using the `>!spoiler!<` (inline) and `>!spoiler` (block) as outlined [here](https://talk.commonmark.org/t/what-could-a-spoiler-tag-extension-look-like/767/86) since it seems they are already widely used by sites similar to Lemmy's form and function like Stack and Reddit, thus people are already familiar with their use very widely.

**Comment by wjs018** [comment]
Just wanted to chime in here on the style convention discussion. PieFed has had in-line spoilers like this for a while and used the telegram convention of `|| spoiler text here ||`. So, if we are looking to maximize compatibility, I would advocate for that style.

**Comment by dessalines** [comment]
If someone wants to add a markdown-it library to support that, go right ahead.

**Comment by DraconicNEO** [comment]
> Just wanted to chime in here on the style convention discussion. PieFed has had in-line spoilers like this for a while and used the telegram convention of `|| spoiler text here ||`. So, if we are looking to maximize compatibility, I would advocate for that style.

Why not support multiple styles?

I mean markdown supports different methods to style like by using underscores or asterisks. So why not use both `||spoiler||`  and `>!spoiler!<` to accomplish the same.

**Comment by wjs018** [comment]
Sure, the discussion seemed to be mostly debating style choices, so I was offering a reason to settle on a convention. However, this is absolutely correct:

> Why not support multiple styles?

So, I just pushed an update for PieFed so that it supports both the `|| telegram/discord ||` style and the `>! reddit !<` style.

I don't think the stack overflow style of marking a whole line is useful as it simply reproduces the existing spoiler block functionality.

**Comment by DraconicNEO** [comment]
> Sure, the discussion seemed to be mostly debating style choices, so I was offering a reason to settle on a convention. However, this is absolutely correct:
> 
> > Why not support multiple styles?
> 
> So, I just pushed an update for PieFed so that it supports both the `|| telegram/discord ||` style and the `>! reddit !<` style.
> 
> I don't think the stack overflow style of marking a whole line is useful as it simply reproduces the existing spoiler block functionality.

I agree whole line of whole paragraph spoiler functionality isn't that useful. Being able to hide individual words or sentences that might span multiple lines is way more useful than hiding a single line.
